bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/util/ShellyChannelCache.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.openhab.binding.shelly.internal.util;
+
+import static org.openhab.binding.shelly.internal.util.ShellyUtils.mkChannelId;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang.Validate;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.core.types.State;
+import org.openhab.binding.shelly.internal.handler.ShellyBaseHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link ShellyChannelCache} implements a caching layer for channel updates.
+ *
+ * @author Markus Michels - Initial contribution
+ */
+@NonNullByDefault
+public class ShellyChannelCache {
+    private final Logger logger = LoggerFactory.getLogger(ShellyChannelCache.class);
+
+    private final ShellyBaseHandler thingHandler;
+    private String thingName = "";
+    private Map<String, Object> channelData = new HashMap<>();
+    private boolean enabled = false;
+
+    public ShellyChannelCache(ShellyBaseHandler thingHandler) {
+        this.thingHandler = thingHandler;
+        this.thingName = thingHandler.thingName;
+    }
+
+    public void setThingName(String thingName) {
+        this.thingName = thingName;
+    }
+
+    public boolean isEnabled() {
+        return enabled;
+    }
+
+    public void enable() {
+        enabled = true;
+    }
+
+    public void disable() {
+        enabled = false;
+    }
+
+    /**
+     * Update one channel. Use Channel Cache to avoid unnecessary updates (and avoid
+     * messing up the log with those updates)
+     *
+     * @param channelId Channel id
+     * @param value Value (State)
+     * @param forceUpdate true: ignore cached data, force update; false check cache of changed data
+     * @return true, if successful
+     */
+    @SuppressWarnings("null")
+    public boolean updateChannel(String channelId, State value, Boolean forceUpdate) {
+        Validate.notNull(channelData);
+        Validate.notNull(channelId);
+        Validate.notNull(value, "updateChannel(): value must not be null!");
```

Comment:
Static analysis should have made these validation checks redundant.
Also, how is channelData supposed to be null here anyhow?

Reply:
removed